crates/matrix-sdk-base/src/sliding_sync.rs

```diff
@@ -199,28 +199,13 @@ impl BaseClient {
         // Handle read receipts and typing notifications independently of the rooms:
         // these both live in a different subsection of the server's response,
         // so they may exist without any update for the associated room.
-
-        for (room_id, raw) in &extensions.receipts.rooms {
-            processors::ephemeral_events::dispatch_one(&mut context, raw.cast_ref(), room_id);
-
-            // We assume this can only happen in joined rooms, or something's very wrong.
-            room_updates
-                .joined
-                .entry(room_id.to_owned())
-                .or_insert_with(JoinedRoomUpdate::default)
-                .ephemeral
-                .push(raw.clone().cast());
-        }
-
-        for (room_id, raw) in &extensions.typing.rooms {
+        processors::room::msc4186::extensions::ephemeral_events(
```

Comment:
I'm still stuck on the naming, I find it strange that there's no verb anywhere in here 😛

Reply:
Rename to `dispatch_ephemeral_events` :-).